forum/utils.py

```diff
@@ -30,3 +33,74 @@ def handle_proxy_requests(request, suffix, method):
         headers=request_headers,
         timeout=5.0,
     )
+
+
+def flag_as_abuse(user, entity):
+    """
+    Flag an entity as abuse.
+
+    Args:
+        user (dict): The user who is flagging the entity as abuse.
+        entity (dict): The entity being flagged as abuse.
+
+    Returns:
+        dict: The updated entity with the abuse flag.
+
+    Raises:
+        ValueError: If user ID or entity is not provided.
+    """
+    if not (user and entity):
+        raise ValueError("User ID/Entity is required")
+    # Check if user ID is already in abuse flaggers list
+    abuse_flaggers = entity["abuse_flaggers"]
+    if user["_id"] not in abuse_flaggers:
+        abuse_flaggers.append(user["_id"])
+        Contents().collection.update_one(
```

Comment:
This is typically mongodb logic that should be in models/, not in a utility function. Remember that at some point we'll want to migrate away from mongodb, so we should not spray our code with mongodb-specific stuff.

Reply:
Done